### PR TITLE
feat(train)!: remove hybrid training mode, rework the training tutorial

### PR DESCRIPTION
**Pull request by WindChimeRan**
## Summary

Two changes, one of them breaking.

### Removes hybrid training mode (breaking)

`--on-generate cache` kept freshly generated hidden states so later epochs could reuse them. It was broken on the Mooncake backend: `MooncakeTransfer` never overrode `cache()`, so it inherited the no-op hook on `HiddenStatesTransfer`, and because the cache branch bypassed the delete path entirely, every generated sample leaked into the store while nothing was ever cached — `MooncakeTransfer.get_cached()` returns `None` unconditionally, so each epoch regenerated everything from scratch.

`--on-generate` now accepts only `delete`, which has always been its default, so existing command lines are unaffected and `--on-generate cache` fails validation with a clear message. The flag is kept purely for backwards compatibility. The now-unused `cache()` hook is dropped from the `HiddenStatesTransfer` API and from `FileTransfer`.

Online and offline training are unchanged.

### Makes the training tutorial runnable end to end

- starts with the public [`inference-optimization/speculators-ci-datasets`](https://huggingface.co/datasets/inference-optimization/speculators-ci-datasets) `tutorial_regen` split, 5K on-policy Qwen3-8B rows -- the same dataset the e2e acceptance tests already use;
- replaces unrelated reference numbers with smoke-test results produced by the documented commands;
- records measured generation time and disk use, documents `hf:<id>[:<subset>:<split>]` inputs, and fixes the DFlash serving command;
- drops the hybrid walkthrough, matching the removal above;
- aligns the remaining DFlash2 and multi-node commands with the unified CLI from #1024, which had missed them.

## 5k tutorial results

4 H100s, `tutorial_regen`, 5 epochs, `checkpoint_best`, evaluated across all nine `RedHatAI/speculator_benchmarks` subsets:

| Algorithm | Draft tokens per step | Acceptance length | Position-0 acceptance | Training time |
| --------- | --------------------- | ----------------- | --------------------- | ------------- |
| Eagle-3   | 3                     | 1.89              | 52.4%                 | 15 min        |
| DFlash    | 15                    | 1.99              | 57.9%                 | 19 min        |
| DFlash2   | 7                     | 1.90              | 52.1%                 | 20 min        |
| DSpark    | 8                     | 2.02              | 54.2%                 | 17 min        |

### CODE REVIEW

**Review by fynnsu — COMMENTED** [review]
> Dataset ownership. The dataset currently lives in a personal HF namespace. Happy to move it under RedHatAI/ or vllm-project/ if maintainers prefer; that changes only the URL in Step 1.

Yeah we should probably move this to RedHatAI/ HF account. 

> Epochs 5 -> 10 matches the measured runs. checkpoint_best still selects on validation loss, so for DFlash (which bottoms out around epoch 4 on this data) the extra epochs cost time, not quality.

I'd prefer we keep this at 5 (or even 3 epochs). At just 5k samples we're definitely overfitting and I think keeping the tutorial short/easy to run is more important than a slight improvement in final acc rates. We just need to make it clear this is just an example and real runs require more data.

**Comment by WindChimeRan** [comment]
@fynnsu Thanks for the review! 
* I'll rerun the training on 5 epoch. 
* Could you please help me with the dataset donation? I think I don't have the permission from the redhat side. Or maybe you can just download the dataset and re-upload to redhat. (then I'll change the dataset name in this PR)

**Comment by WindChimeRan** [comment]
@fynnsu ready for review!
new changes
* dataset now use the CI hf path
* remove the hybrid mode training code
* rerun the 5 epoch training for all models, including the new dflash2
* training tutorial now include both data paths (regen and renderer endpoint)

File: src/speculators/train/config/schema.py [hunk 1]
```diff
@@ -270,11 +270,11 @@ class GenerationArgs(_Group):
         "Default 'generate' generates them on demand via the vLLM endpoint; the others "
         "skip the sample, skip with a warning, or raise.",
     )
-    on_generate: Literal["cache", "delete"] = Field(
+    on_generate: Literal["delete"] = Field(
```
Comment (fynnsu) [inline src/speculators/train/config/schema.py:273]: Maybe we should raise a deprecation notice when the field is passed in explicitly. Then we could remove the field in a release or two.

File: docs/user_guide/tutorials/train.md [hunk 2]
```diff
@@ -819,52 +686,38 @@ Check vLLM logs for speculative decoding metrics.
...
 # avg_seq_len × num_layers × hidden_size × dtype_bytes
-# 8192 × 4 × 4096 × 2 = ~268 MB per sample
+# 3000 × 4 × 4096 × 2 = ~98 MB per sample   (Qwen3-8B, 4 captured layers)
 
-# For 50K samples: ~13 TB
-# For 10K samples: ~2.6 TB
-# For 1K samples: ~260 GB
+# For 5K samples: ~460 GB
+# For 10K samples: ~960 GB
+# For 50K samples: ~4.8 TB
```
Comment (fynnsu) [inline docs/user_guide/tutorials/train.md]: How were these obtained? Like the first one I get is measure. But then should 10k be 920 GB? and 50K 4.6 TB?

File: docs/user_guide/tutorials/train.md
Comment (fynnsu) [inline docs/user_guide/tutorials/train.md:1]: I'd prefer to split this out into a separate pr because it seems independent from the hybrid training work.

**Review by fynnsu — COMMENTED** [review]
Hi Ranran, thanks for the pr!

File: src/speculators/train/config/schema.py (hunk 1 above)
Comment (WindChimeRan) [reply src/speculators/train/config/schema.py:273]: Before adding the warning I want to check the bigger picture with you, because the UX becomes more spaghetti now. 

**How the modes are selected**

| | Online | Offline | Hybrid |
|---|---|---|---|
| Before this PR | `--on-missing generate --on-generate delete` | `--on-missing raise` | `--on-missing generate --on-generate cache` |
| This PR | `--on-missing generate` | `--on-missing raise` | removed |
| Ideally | `--training-mode online` | `--training-mode offline` | — |

File: docs/user_guide/tutorials/train.md (hunk 2 above)
Comment (WindChimeRan) [reply docs/user_guide/tutorials/train.md]: done

File: docs/user_guide/tutorials/train.md
Comment (WindChimeRan) [reply docs/user_guide/tutorials/train.md:1]: I prefer 1 PR. because split will result in unnecessary merge conflicts. 

The change is small.
